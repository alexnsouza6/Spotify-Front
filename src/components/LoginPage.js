import React from "react";
import { Button, Grid, Segment, Header, Image } from "semantic-ui-react";
import "../common/stylesheets/container.css";
import "../common/stylesheets/button.css";

const LoginPage = () => (
  <Grid
    verticalAlign="middle"
    textAlign="center"
    className="container-fsize container-color--black"
  >
    <Grid.Column>
      <Segment className="container-hsize container-color--black">
        <Header as="h1" size="huge" style={{ color: "white" }}>
          <Image circular src={require("../common/images/spotify.png")} />
          Which artists are you following on Spotify?
        </Header>
        <Button
          className="button-rounded button-style--default"
          as="a"
          href="https://spotify-api-artists-only.herokuapp.com/api/v1/login"
        >
          FIND OUT{" "}
        </Button>
      </Segment>
    </Grid.Column>
  </Grid>
);

export default LoginPage;
