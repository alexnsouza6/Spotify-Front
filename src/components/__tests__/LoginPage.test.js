import React from "react";
import { render } from "enzyme";
import renderer from "react-test-renderer";
import LoginPage from "../LoginPage";
import "../../setupTests";

describe("when testing w/ snapshots", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<LoginPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("when testing html attributes", () => {
  it("renders a button with specific href", () => {
    const component = render(<LoginPage />);
    const button = component.find("a");
    const redirectURL = button[0].attribs.href;
    expect(redirectURL).toEqual(
      "https://spotify-api-artists-only.herokuapp.com/api/v1/login"
    );
  });
});
