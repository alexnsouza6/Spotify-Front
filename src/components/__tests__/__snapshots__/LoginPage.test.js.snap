// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when testing w/ snapshots renders correctly 1`] = `
<div
  className="ui center aligned middle aligned grid container-fsize container-color--black"
>
  <div
    className="column"
  >
    <div
      className="ui segment container-hsize container-color--black"
    >
      <h1
        className="ui huge header"
        style={
          Object {
            "color": "white",
          }
        }
      >
        <img
          className="ui circular image"
          src="spotify.png"
        />
        Which artists are you following on Spotify?
      </h1>
      <a
        className="ui button button-rounded button-style--default"
        href="https://spotify-api-artists-only.herokuapp.com/api/v1/login"
        onClick={[Function]}
        role="button"
      >
        FIND OUT
         
      </a>
    </div>
  </div>
</div>
`;
